forge-game/src/main/java/forge/game/cost/Cost.java

```diff
@@ -883,7 +883,22 @@ public static String convertAmountTypeToWords(final String amount, final String
         return sb.toString();
     }
 
+    public void mergeTo(Cost source, int amt) {
+        if (source.isOnlyManaCost()) {
+            // multiply cost
+            for (int i = 0; i < amt; ++i) {
+                this.add(source);
+            }
+        } else {
+            source.getCostParts().get(0).setAmount(Integer.toString(amt));
```

Comment:
* Jötun Grunt Cumulative upkeep—Put two cards from a single graveyard on the bottom of their owner’s library.
* Infernal Darkness Cumulative upkeep—Pay {B} and 1 life.

* Wall of Shards Cumulative upkeep—An opponent gains 1 life. 
> When paying Wall of Shards’ cumulative upkeep, you may choose a different opponent for each age counter, or you can choose the same opponent multiple times.

Reply:
Right, no shortcut then!